server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/reactivestreams/FileWriterSubscriber.scala

```diff
@@ -0,0 +1,61 @@
+package sttp.tapir.server.netty.internal.reactivestreams
+
+import io.netty.handler.codec.http.HttpContent
+import org.reactivestreams.{Publisher, Subscription}
+
+import java.nio.channels.AsynchronousFileChannel
+import java.nio.file.{Path, StandardOpenOption}
+import scala.concurrent.{Future, Promise}
+
+class FileWriterSubscriber(path: Path) extends PromisingSubscriber[Unit, HttpContent] {
+  private var subscription: Subscription = _
+  private var fileChannel: AsynchronousFileChannel = _
+  private var position: Long = 0
+  private val resultPromise = Promise[Unit]()
+
+  override def future: Future[Unit] = resultPromise.future
+
+  override def onSubscribe(s: Subscription): Unit = {
+    this.subscription = s
+    fileChannel = AsynchronousFileChannel.open(path, StandardOpenOption.WRITE, StandardOpenOption.CREATE)
+    s.request(1)
+  }
+
+  override def onNext(httpContent: HttpContent): Unit = {
+    val byteBuffer = httpContent.content().nioBuffer()
+    fileChannel.write(
+      byteBuffer,
+      position,
+      (),
+      new java.nio.channels.CompletionHandler[Integer, Unit] {
+        override def completed(result: Integer, attachment: Unit): Unit = {
+          position += result
```

Comment:
this should be volatile at least -we've got a mutable variable accessed from multiple threads

Reply:
ah, so here we only ever demand single elements, so that's our protection agains `complete` and `onNext` getting called concurrently? if so, a comment would be nice :)

Reply:
Yes, with request(1) we don't need to handle receiving `onNext` before the previous write finishes. However, I'm worried that this is pretty bad for performance.

Reply:
we have to measure it :) there's some very basic infrastructure for the in `perf-tests`, not sure if the tests use a body, though

Reply:
still, even if things are ever called on a single thread, these might be different threads - so `position` should be volatiole. And another follow-up task: performance of writing files